pyrdp/ui/qt.py

```diff
@@ -174,26 +183,40 @@ def setMousePosition(self, x: int, y: int):
         self.mouseY = y
         self.update()
 
+    def scale(self, scale):
+        """
+        Rescale the current widget to a percentage of the height of the RDP session.
+        :param scale: Ex: 0.5 for 50% height and 50% width.
+        """
+        self.scaleValue = scale
+        self.resize(self.sessionWidth, self.sessionHeight)
```

Comment:
No.

The rendering surface that the drawing pipeline use must remain the full resolution of RDP, since all bitmap coords and offset are based on that resolution. Scaling should only happen at the very last moment before painting.

Reply:
see last comment